Lib9c/Module/AgentModule.cs

```
@@ -21,7 +22,16 @@ public static class AgentModule
 
             try
             {
-                return new AgentState((Bencodex.Types.Dictionary)serializedAgent);
+                if (serializedAgent is Dictionary dict)
+                {
+                    return new AgentState(dict);
+                }
+                else if (serializedAgent is List list)
+                {
+                    return new AgentState(list);
+                }
+
+                return null;
```

Throwing exception instead of returning null would be better for L34 IMHO.